server/src/main/java/org/opensearch/search/aggregations/bucket/FilterRewriteHelper.java

```diff
@@ -252,8 +304,11 @@ static boolean tryFastFilterAggregation(
                     ),
                     counts[i]
                 );
+                s++;
+                if (s > size) return true;
             }
         }
-        throw new CollectionTerminatedException();
+
+        return true;
```

Comment:
Any specific reason for not throwing the exception?

Reply:
Moving this tryFastFilterAggregation method up a bit from collect method to getLeafCollector. 
So this is run when leaf collector being constructed, to make it more flexible and readable in Aggregator getLeafCollector, throw the `CollectionTerminatedException` there, indicating if optimized, we stop the following collect.